torch/_dynamo/external_utils.py

```diff
@@ -27,3 +27,20 @@ def call_hook(hook, *args):
     if result is None:
         return args[0]
     return result
+
+
+class FakeContext:
+    def __init__(self, saved_tensors):
+        # this will cache the results of saved_tensors
+        # and will no longer call into c++ binding
+        self.saved_tensors = saved_tensors
+
+
+def call_backward(backward_fn, saved_tensors, *args):
+    grads = backward_fn(FakeContext(saved_tensors), *args)
+
+    # in eager, we wrap in a tuple when there's only one grad output
+    if type(grads) is not tuple:
```

Comment:
Double check handling when return type is `list`

Reply:
eager behavior is to wrap the `list` in a tuple too: https://github.com/pytorch/pytorch/blob/main/torch/csrc/autograd/python_function.h#L62